DRAFT.md

```diff
@@ -216,15 +216,80 @@ Note: As the null state of `indirect` and `polymorphic` is not observable, and
 access to a moved-from object is erroneous, `std::optional` can be specialized
 by implementers to exchange pointers on move construction and assignment.
 
-### Noexcept
+### Allocators as template arguments
 
 TODO
 
-### Allocators as template arguments
+### Modelled types
+
+The class templates `indirect` and `polymorphic` have strong similarities to
+existing class templates. These similarities motivate much of the design, we aim
+for consistency with existing library types, not innovation.
+
+#### Modelled types for `indirect`
+
+The class template `indirect` owns an object of known type, permits copies,
+propagates const and is allocator aware.
+
+Like `optional` and `unique_ptr`, `indirect` can be in a valueless state;
```

Comment:
Like `optional` and `unique_ptr`, `indirect` can be in a valueless state. `indirect` can only get into the valueless state after move.